clicommand/pipeline_upload.go

```diff
@@ -287,14 +308,47 @@ var PipelineUploadCommand = cli.Command{
 			}
 		}
 
-		// In dry-run mode we just output the generated pipeline to stdout
+		if cfg.SigningKeyFile != "" {
```

Comment:
it might be good to add a dragons warning here - "this is experimental and will change! it might not work!" etc